### DISCUSSION TITLE
Slow deserialization

### DISCUSSION THREAD

**Discussion by jianyilim** [OData/odata.net#2697] [General]
The  OData service promptly delivered the response within 100ms. However, an additional delay of close to 10 seconds was observed during the completion of the ExecuteAsync method. It appears that this delay could be attributed to the deserialization process. I am seeking guidance on how to investigate and determine the root cause of this slowness.

**Comment by habbes** [comment]
Hi @jianyilim could you share a code sample? Profiling your application could help. If you're sure that the client receives the response from the server within 100ms, then CPU profiling can help determine what functions the processor is spending most of the time on. If the delay is caused by some I/O latency (e.g. another network call, blocked threads, etc.) then other profiling techniques can help determine this.

If you're using Visual Studio, you can follow this guide to learn more about the profiling tools that it ships with: https://learn.microsoft.com/en-us/visualstudio/profiling/profiling-feature-tour?view=vs-2022. Visual Studio includes various profiling tools for investigating different kinds of issues: CPU usage, memory usage and allocations, asynchronous code, etc.
